3604/1/rating_creator.rb

```diff
@@ -0,0 +1,25 @@
+# class for create rating
+class RatingCreator
+  HASH_COEFFICIENTS = {
+    used_by: 1,
+    watched_by: 0.5,
+    stars: 1,
+    forks: 0.7,
+    contributors: 0.5,
+    issues: 0.1
+  }.freeze
+
+  def array_of_rating_gems(array_of_all_gems)
+    array_of_all_gems.sort_by { |hash_of_one_gem| -sum_of_gem(hash_of_one_gem) }
+  end
+
+  private
+
+  def sum_of_gem(hash_of_one_gem)
+    hash_of_one_gem = hash_of_one_gem.map do |key, value|
```

Comment:
1. map will return array so it is not a hash (naming issue)
2. I don't like line 22. It's good that you explain why you have such a confusing line but better to have self-explanatory code. I suggest you to refactor method: please iterate over `HASH_COEFFICIENTS` instead of `hash_of_one_gem `,  it will help to avoid condition in the block (because `HASH_COEFFICIENTS` contains all keys you need and doesn't contains `name` which previously leaded to nil value in resulting array)
3. Strange method name. Is it `rating_of_gem`?